traitsui/tests/editors/test_set_editor.py

```diff
@@ -0,0 +1,592 @@
+            items.append(list_widget.item(i).text())
+
+    return items
+
+
+def click_on_item(editor, item_no, in_used=False):
+    """ Simulate a click on an item in a specified list. """
+    unused_list = editor._unused
+    used_list = editor._used
+
+    if is_current_backend_wx():
+        import wx
+
+        # First deselect all items
+        for i in range(unused_list.GetCount()):
+            unused_list.Deselect(i)
+        for i in range(used_list.GetCount()):
+            used_list.Deselect(i)
+        # Select the item in the correct list
+        list_with_selection = used_list if in_used else unused_list
+        list_with_selection.SetSelection(item_no)
+
+        event = wx.CommandEvent(
+            wx.EVT_LISTBOX.typeId, list_with_selection.GetId()
+        )
+        wx.PostEvent(editor.control, event)
+
+    elif is_current_backend_qt4():
+        for i in range(unused_list.count()):
+            status = (not in_used) and (item_no == i)
+            unused_list.item(i).setSelected(status)
+
+        for i in range(used_list.count()):
+            status = (in_used) and (item_no == i)
+            used_list.item(i).setSelected(status)
+
+        if in_used:
+            used_list.itemClicked.emit(used_list.item(item_no))
+        else:
+            unused_list.itemClicked.emit(unused_list.item(item_no))
+
+
+def double_click_on_item(editor, item_no, in_used=False):
+    """ Simulate a double click on an item in a specified list. """
+    unused_list = editor._unused
+    used_list = editor._used
+
+    if is_current_backend_wx():
+        import wx
+
+        # First deselect all items
+        for i in range(unused_list.GetCount()):
+            unused_list.Deselect(i)
+        for i in range(used_list.GetCount()):
+            used_list.Deselect(i)
+        # Select the item in the correct list
+        list_with_selection = used_list if in_used else unused_list
+        list_with_selection.SetSelection(item_no)
+
+        event = wx.CommandEvent(
+            wx.EVT_LISTBOX_DCLICK.typeId, list_with_selection.GetId()
+        )
+        wx.PostEvent(editor.control, event)
+
+    elif is_current_backend_qt4():
+        for i in range(unused_list.count()):
+            status = (not in_used) and (item_no == i)
+            unused_list.item(i).setSelected(status)
+
+        for i in range(used_list.count()):
+            status = (in_used) and (item_no == i)
+            used_list.item(i).setSelected(status)
+
+        if in_used:
+            used_list.itemDoubleClicked.emit(used_list.item(item_no))
+        else:
+            unused_list.itemDoubleClicked.emit(unused_list.item(item_no))
+
+
+def get_button_enabled(button):
+    """ Return if the button is enabled or not."""
+    if is_current_backend_wx():
+        return button.IsEnabled()
+
+    elif is_current_backend_qt4():
+        return button.isEnabled()
+
+
+def click_button(button):
+    """ Click the provided button. """
```

Comment:
This function is general enough that it will be useful for other tests too.
I think there are at least two similar instances in the code base already. Could this be contributed to somewhere more reusable e.g. `traitsui.tests._tools`?